### PR TITLE
Cleaning up DpdkTestpmd.

### PR DESCRIPTION
**Pull request by mcgov**
- remove old assert wrapper function, use expected_exit_code and add more useful failure tips.
- cleanup some lingering RHEL issues

### CODE REVIEW

**Comment by mcgov** [comment]
Hold for review, do not merge. After the code to fix find_parition_with_freespace merges I will change this to check that there is space for the DPDK installation as well since it is large.

File: microsoft/testsuites/dpdk/dpdktestpmd.py [hunk 1]
```diff
@@ -416,39 +446,122 @@ def _install_dependencies(self) -> None:
...
-            self.__execute_assert_zero(
-                "yum update -y --disablerepo='*' --enablerepo='*microsoft*'", cwd
-            )
+            if node.os.information.version.major < 7:
+                raise UnsupportedDistroException(
+                    node.os, "DPDK for Redhat < 7 is not supported by this test"
+                )
+            elif node.os.information.version.major == 7:
+                # Add packages for rhel7
+                node.os.install_packages(list(["libmnl-devel", "libbpf-devel"]))
```
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdktestpmd.py:499]: Add else or comments to explain what happened to other versions.

File: microsoft/testsuites/dpdk/dpdktestpmd.py [hunk 2]
```diff
@@ -416,39 +446,122 @@ def _install_dependencies(self) -> None:
...
-            # TODO: Workaround for type checker below
             node.os.install_packages(list(self._redhat_packages))
-
-            self.__execute_assert_zero("systemctl enable rdma", cwd)
-            self.__execute_assert_zero("pip3 install --upgrade meson", cwd)
-            self.__execute_assert_zero("ln -s /usr/local/bin/meson /usr/bin/meson", cwd)
-
+            # NOTE: Some versions need this and some don't for a bunch of
+            # different reasons. Total hack: run it and don't check the exit code.
+            node.execute(
```
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdktestpmd.py]: Use the service tool. It combines all service components together.

**Comment by mcgov** [comment]
ready for review

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**